dns_sd.c

```diff
@@ -32,6 +32,13 @@
 #define close(s) closesocket(s)
 #endif
 
+static void dnssd_free_discovery_data(struct dns_sd_discovery_data *d)
```

Comment:
Did you test this on a Mac? 

While this is the same as the existing Linux code, it's different than the bonjour code. (which is why it was separate before).

Reply:
`d->address` is NULL everywhere except on Linux, and `free(NULL)` is defined by the C standard to be a NOP, so this shouldn't break anything.

I did not test it on Mac, maybe @AlexandraTrifan can verify that it still works fine?

Reply:
Alexandra tested it, and it still works fine.